.github/workflows/deploy.yml

```diff
@@ -0,0 +1,71 @@
+          docker tag dashboard:latest ${{ secrets.ECR_DASHBOARD }}:latest
+          docker push ${{ secrets.ECR_DASHBOARD }}:latest
+
+      - name: Build and Push Quadhoral-Openmeteo Docker Image
+        if: contains(github.event.pull_request.head.ref, 'quadhoral-openmeteo') && (hashFiles('quadhoral-openmeteo/Dockerfile') != '')
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_QUADHORAL_OPENMETEO }}
+          docker build -t quadhoral-openmeteo:latest ./quadhoral-openmeteo
+          docker tag quadhoral-openmeteo:latest ${{ secrets.ECR_QUADHORAL_OPENMETEO }}:latest
+          docker push ${{ secrets.ECR_QUADHORAL_OPENMETEO }}:latest
+
+      - name: Build and Push Weekly-Openmeteo Docker Image
+        if: contains(github.event.pull_request.head.ref, 'weekly-openmeteo') && (hashFiles('weekly-openmeteo/Dockerfile') != '')
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_WEEKLY_OPENMETEO }}
+          docker build -t weekly-openmeteo:latest ./weekly-openmeteo
+          docker tag weekly-openmeteo:latest ${{ secrets.ECR_WEEKLY_OPENMETEO }}:latest
+          docker push ${{ secrets.ECR_WEEKLY_OPENMETEO }}:latest
+
+      - name: Build and Push Astronomy Pipeline Docker Image
+        if: contains(github.event.pull_request.head.ref, 'astronomy-pipeline') && (hashFiles('astronomy-pipeline/Dockerfile') != '')
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_ASTRONOMY_PIPELINE }}
+          docker build -t astronomy-pipeline:latest ./astronomy-pipeline
+          docker tag astronomy-pipeline:latest ${{ secrets.ECR_ASTRONOMY_PIPELINE }}:latest
+          docker push ${{ secrets.ECR_ASTRONOMY_PIPELINE }}:latest
+
+      - name: Build and Push Aurorawatch Pipeline Docker Image
+        if: contains(github.event.pull_request.head.ref, 'aurorawatch-pipeline') && (hashFiles('aurorawatch-pipeline/Dockerfile') != '')
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_AURORAWATCH_PIPELINE }}
+          docker build -t aurorawatch-pipeline:latest ./aurorawatch-pipeline
+          docker tag aurorawatch-pipeline:latest ${{ secrets.ECR_AURORAWATCH_PIPELINE }}:latest
+          docker push ${{ secrets.ECR_AURORAWATCH_PIPELINE }}:latest
+
+      - name: Build and Push SMS-Checker Docker Image
+        if: contains(github.event.pull_request.head.ref, 'sms-checker') && (hashFiles('sms-checker/Dockerfile') != '')
+        run: |
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | docker login --username AWS --password-stdin ${{ secrets.ECR_SMS_CHECKER }}
+          docker build -t sms-checker:latest ./sms-checker
```

Comment:
Docker build, push and tag are reused several times across this shell script. Would it be worth making a matrix that can iteratively apply these docker actions? This will work as is, but this could be a potential improvement.

Reply:
This is a good idea for a future improvement